wallet/txauthor/author.go

```diff
@@ -77,12 +77,11 @@ type ChangeSource func() ([]byte, error)
 // InputSourceError is returned.
 //
 // BUGS: Fee estimation may be off when redeeming non-compressed P2PKH outputs.
-// TODO(roasbeef): fix fee estimation for witness outputs
 func NewUnsignedTransaction(outputs []*wire.TxOut, relayFeePerKb btcutil.Amount,
 	fetchInputs InputSource, fetchChange ChangeSource) (*AuthoredTx, error) {
 
 	targetAmount := h.SumOutputValues(outputs)
-	estimatedSize := txsizes.EstimateSerializeSize(1, outputs, true)
+	estimatedSize := txsizes.EstimateVirtualSize(1, 0, 0, outputs, true)
```

Comment:
Seems like this will always overshoot if it assumes that we _always_ have a single p2pkh input? With my pending PR, the wallet will never even created p2pkh addresses for the end user. 

Before the commit prior to this one, the only argument was the number of inputs (assumed to be one).

Reply:
So it's the _second_ argument that should be on. It's the case that this is just used for an initial estimate though, as the loops below will start to create a bit of a tighter bound.

Reply:
I agree, it will give us a lower initial estimate. 

By the same argument, I wanted to set `changeOutput=false`, but that turned out to complicate the following fee estimation significantly...